types/react-native/index.d.ts

```diff
@@ -7000,12 +7000,17 @@ export type BackPressEventName = 'hardwareBackPress';
  * Detect hardware back button presses, and programmatically invoke the
  * default back button functionality to exit the app if there are no
  * listeners or if none of the listeners return true.
- * Methods don't have more detailed documentation as of 0.25.
+ * The event subscriptions are called in reverse order
+ * (i.e. last registered subscription first), and if one subscription
+ * returns true then subscriptions registered earlier
+ * will not be called.
+ *
+ * @see https://reactnative.dev/docs/backhandler.html
  */
 export interface BackHandlerStatic {
     exitApp(): void;
-    addEventListener(eventName: BackPressEventName, handler: () => void): NativeEventSubscription;
-    removeEventListener(eventName: BackPressEventName, handler: () => void): void;
+    addEventListener(eventName: BackPressEventName, handler: () => true | void): NativeEventSubscription;
+    removeEventListener(eventName: BackPressEventName, handler: () => true | void): void;
```

Comment:
This change seems incorrect.. Even without this change, you should be able to pass handle that returns true

Reply:
Yes, indeed, one is already able to pass a handler that returns anything at all, as no return value is strictly needed. However the handler must return `true`, as per documentation (or actually any truthy value, as per implementation), to cause specific behaviour - stopping all the subsequent listeners, including the default one, from executing. And it seems appropriate to document that the handler "should" return `true` or nothing (even though, in the implementation, the return value is typecasted to boolean, and any value is possible).

Reply:
[The Flow version](https://github.com/facebook/react-native/blob/8553e1acc4195479190971cc7a3ffaa0ed37a5e0/Libraries/Utilities/BackHandler.ios.js#L57) doesn’t really say anything about return type. Looking at the docs and the implementation, it seems to me like this should just be `() => boolean`, but perhaps `() => boolean | null | undefined` is acceptable too.

In any case, it would be good to send a similar change upstream, unless I’m overlooking a reason for this to not be typed as such already. /cc @TheSavior

Reply:
I’m not familiar with this. If the flow code doesn’t include boolean then it is likely that we don’t have any callsites doing that, which makes me not trust the documentation. Can you verify the runtime behavior of the listeners and behavior of returning true?

Reply:
Yes, the links are attached
- https://github.com/facebook/react-native/blob/8553e1acc4195479190971cc7a3ffaa0ed37a5e0/Libraries/Utilities/BackHandler.android.js#L24
- https://github.com/facebook/react-native/blob/8553e1acc4195479190971cc7a3ffaa0ed37a5e0/Libraries/Utilities/BackHandler.ios.js#L79

Reply:
Ah, okay, I understand better now. The flow code for those functions is unruled. Function is the same as any. I think having the type be boolean | void is reasonable. Please upstream a PR making it typed as returning ?boolean.

Reply:
> I think having the type be boolean | void is reasonable. Please upstream a PR making it typed as returning ?boolean.

I _assume_ you meant the Flow equivalent that I posted: `() => boolean | null | undefined`. Using `void` would I think send the wrong message, i.e. that the return value would not lead to effects.

Reply:
Correct. I was agreeing with what you posted :-) Thanks for checking